driver/ni/ai_channels.h

```diff
@@ -0,0 +1,1503 @@
+//                                      Temperature                              //
+///////////////////////////////////////////////////////////////////////////////////
+class Thermocouple final : public Analog {
+public:
+    int32_t thermocoupleType;
+    int32_t cjcSource;
+    double cjcVal;
+    std::string cjcChannel;
+
+    static int32_t getType(std::string type) {
+        if (type == "J") return DAQmx_Val_J_Type_TC;
+        if (type == "K") return DAQmx_Val_K_Type_TC;
+        if (type == "N") return DAQmx_Val_N_Type_TC;
+        if (type == "R") return DAQmx_Val_R_Type_TC;
+        if (type == "S") return DAQmx_Val_S_Type_TC;
+        if (type == "T") return DAQmx_Val_T_Type_TC;
+        if (type == "B") return DAQmx_Val_B_Type_TC;
+        if (type == "E") return DAQmx_Val_E_Type_TC;
+
+        LOG(ERROR) << "Invalid TC Type";
+        return DAQmx_Val_J_Type_TC;
+    }
+
+    static int32_t getCJCSource(std::string source) {
+        if (source == "BuiltIn") return DAQmx_Val_BuiltIn;
+        if (source == "ConstVal") return DAQmx_Val_ConstVal;
+        if (source == "Chan") return DAQmx_Val_Chan;
+        LOG(ERROR) << "Invalid cjc type";
+        return DAQmx_Val_BuiltIn;
+    }
+
+
+    explicit Thermocouple(config::Parser &parser, TaskHandle task_handle,
+                          std::string name)
+        : Analog(parser, task_handle, name),
+          thermocoupleType(getType(parser.required<std::string>("thermocouple_type"))),
+          cjcSource(getCJCSource(parser.required<std::string>("cjc_source"))),
+          cjcVal(parser.required<double>("cjc_val")) {
+            std::string u = parser.optional<std::string>("units", "DegC");
+            this->units = ni::UNITS_MAP.at(u);    
+        }
+
+    //cjcChannel(parser.required<std::string>("cjc_channel")) {} FIXME: this property should be take form console
+
+
+    ///	DAQmxErrChk (DAQmxCreateAIThrmcplChan(taskHandle,"","",0.0,100.0,DAQmx_Val_DegC,DAQmx_Val_J_Type_TC,DAQmx_Val_BuiltIn,25.0,""));
+
+    int32 createNIChannel() override {
+        if (this->scale_config->type == "none") {
+            return ni::NiDAQmxInterface::CreateAIThrmcplChan(
+                this->task_handle,
+                this->name.c_str(),
+                "",
+                this->min_val,
+                this->max_val,
+                this->units,
+                this->thermocoupleType,
+                this->cjcSource,
+                this->cjcVal,
+                ""
+            );
+        }
+    }
+};
+
+class TemperatureBuiltInSensor final : public Analog{
+    public:
+        explicit TemperatureBuiltInSensor(config::Parser &parser, TaskHandle task_handle, std::string name){
+            this->task_handle = task_handle;
+            
+            std::string u = parser.optional<std::string>("units", "Volts");
+            this->units = ni::UNITS_MAP.at(u);
+
+            size_t pos = name.find("/");
```

Comment:
Could this cause problems if we don't find a device?

Reply:
We try to find the device before we begin parsing any channels and if that happens and handle the error at that point if we don't find it. So we would not get to this point in a such a case